lib/ex_aws/rds/impl.ex

```diff
@@ -0,0 +1,128 @@
+defmodule ExAws.RDS.Impl do
+
+  @version "2014-10-31"
+
+  def add_source_id_to_subscription(client, source_id, subscription) do
+    query_params = Map.new
+    |> Map.put_new("Action", "AddSourceIdentifierToSubscription")
+    |> Map.put_new("SourceIdentifier", source_id)
+    |> Map.put_new("SubscriptionName", subscription)
+    |> Map.put_new("Version", @version)
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  def add_tags_to_resource(client, resource, tags, n) do
+    query_params = Map.new
+    |> Map.put_new("Action", "AddTagsToResource")
+    |> Map.put_new("ResourceName", resource)
+    |> Map.put_new("Version", @version)
+
+    for {k, v} <- tags do 
+      query_params = query_params
+      |> Map.put_new("Tags.member.#{Integer.to_string(n)}.Key", k)
+      |> Map.put_new("Tags.member.#{Integer.to_string(n)}.Value", v)
+    end
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  def apply_pending_maintenance(client, resource_id, action, opt_in_type) when action == "system-update" do 
+    query_params = Map.new
+    |> Map.put_new("Action", "ApplyPendingMaintenanceAction")
+    |> Map.put_new("ResourceIdentifier", resource_id)
+    |> Map.put_new("ApplyAction", action)
+    |> Map.put_new("OptInType", opt_in_type)
+    |> Map.put_new("Version", @version)
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  def apply_pending_maintenance(client, resource_id, action, opt_in_type) when action == "db-upgrade" do 
+    query_params = Map.new
+    |> Map.put_new("Action", "ApplyPendingMaintenanceAction")
+    |> Map.put_new("ResourceIdentifier", resource_id)
+    |> Map.put_new("ApplyAction", action)
+    |> Map.put_new("OptInType", opt_in_type)
+    |> Map.put_new("Version", @version)
+
+    request(client, :post, "/", params: query_params)
+  end  
+
+  def describe_db_instances(client, opts \\ %{}) do 
+    query_params = Map.new
+    |> Map.put_new("Action", "DescribeDBInstances")
+    |> Map.put_new("Version", @version)
+    |> Map.merge(opts)
+
+    request(client, :get, "/", params: query_params)
+  end
+
+  def create_db_instance(client, instance_id, username, password, storage, class, engine, opts \\ %{}) do 
+    query_params = Map.new
+    |> Map.put_new("Action", "CreateDBInstance")
+    |> Map.put_new("MasterUsername", username)
+    |> Map.put_new("MasterUserPassword", password)
+    |> Map.put_new("AllocatedStorage", storage)
+    |> Map.put_new("DBInstanceIdentifier", instance_id)
+    |> Map.put_new("DBInstanceClass", class)
+    |> Map.put_new("Engine", engine)
+    |> Map.put_new("Version", @version)
+    |> Map.merge(opts)
```

Comment:
There is an additional issue. Right now you simply merge in the `opts`, but this requires that people use `"Engine"` or whatever as keys in the map. One of the more useful things about `ExAws` is that I allow folks to use `db_instance_identifier: "foo"` atom style keys instead of the binary keys that get sent to AWS. See the other clients for details. There are quite a few utility functions that already exist for transforming atom keys given in options to binary keys.